api/src/infrastructure/postgres-survey-data.repository.ts

```diff
@@ -0,0 +1,107 @@
+import { DataSource, EntityManager } from 'typeorm';
+import { ISurveyDataRepository } from '@api/infrastructure/survey-data-repository.interface';
+import { Inject, Logger } from '@nestjs/common';
+import { InjectDataSource } from '@nestjs/typeorm';
+import { WidgetDataFilters } from '@shared/dto/widgets/widget-data-filter';
+import { SectionWithDataWidget } from '@shared/dto/sections/section.entity';
+import { SQLAdapter } from '@api/infrastructure/sql-adapter';
+import { WidgetQuestionMap } from '@shared/dto/widgets/widget-question-map';
+import {
+  BaseWidgetWithData,
+  WidgetData,
+} from '@shared/dto/widgets/base-widget-data.interface';
+
+export class PostgresSurveyDataRepository implements ISurveyDataRepository {
+  public constructor(
+    private readonly logger: Logger,
+    @Inject(SQLAdapter) private readonly sqlAdapter: SQLAdapter,
+    @InjectDataSource() private readonly dataSource: DataSource,
+  ) {}
+
+  public async addSurveyDataToSections(
+    sections: SectionWithDataWidget[],
+    filters?: WidgetDataFilters,
+  ): Promise<SectionWithDataWidget[]> {
+    // Perform a transaction to ensure isolation
+    return await this.dataSource.transaction('READ COMMITTED', async (t) => {
+      let answersTable: string = 'survey.answers';
```

Comment:
This is evil, the datasource has no knowledge of an additional schema, which is created in a sql file, which is again, evil AF. Readability and maintainability of this code is not easy